src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -159,7 +159,14 @@ public IChannelHandle GetChannel(CancellationToken cancellationToken)
                     {
                         // ignore exceptions
                     }
-
+                }
+                try
+                {
+                    HandlePreHandshakeException(ex);
```

Comment:
HandleHandshakeException? I think the name with `pre`doesn't fully described the spec lines

Reply:
Perhaps `HandleBeforeHandshakeCompletesException`? So that it matches the "before handshake completes" section [here](https://github.com/mongodb/specifications/commit/bfddbd98bf396104f78500f93f1f13e3fd2d8c3a#diff-62b4da2e38e7d2a9c1a2d92d315895bbR1245). I typically find having similar names to the spec to be helpful :)  What do you think?

Reply:
Sgtm (I also thought about this naming :) )

Reply:
Great minds think alike!  (⌐■_■)
Done!